oidc_middleware.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"github.com/gin-gonic/gin"
+	"math/big"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+)
+
+// Largely taken from https://developer.okta.com/blog/2021/01/04/offline-jwt-validation-with-go
```

Comment:
This code is licensed under Apache-2.0, so this file needs their license notice and a note stating that you've changed it

Such as:

```suggestion
// This is a modified version of the code provided in the follow blog post and
// GitHub repository:
// - https://developer.okta.com/blog/2021/01/04/offline-jwt-validation-with-go
// - https://github.com/oktadev/okta-offline-jwt-validation-example/tree/a61cc73bf893686c1efe67ce86448047205826bc
//
// Copyright 2019 Okta, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
```